util/src/main/java/com/farao_community/farao/util/FaraoNetworkPool.java

```diff
@@ -27,25 +28,31 @@
 public class FaraoNetworkPool extends ForkJoinPool implements AutoCloseable {
     private static final Logger LOGGER = LoggerFactory.getLogger(FaraoNetworkPool.class);
     protected final BlockingQueue<Network> networksQueue;
-    private final String targetVariant;
+    protected final String targetVariant;
+    protected final String workingVariant;
 
     // State used to save initial content of target variant.
     // Useful when targetVariant equals VariantManagerConstants.INITIAL_VARIANT_ID
-    private final String stateSaveVariant;
+    protected final String stateSaveVariant;
 
-    public FaraoNetworkPool(Network network, String targetVariant, int parallelism) {
+    public static FaraoNetworkPool create(Network network, String targetVariant, int parallelism) {
+        if (parallelism == 1) {
+            return new SingleNetworkPool(network, targetVariant);
+        } else {
+            return new FaraoNetworkPool(network, targetVariant, parallelism);
+        }
+    }
```

Comment:
the design look quite strange for me, in which a class is using the constructor of one of its subclasses.

Can we have an abstract class FaraoNetworkPool and two subclasses SingleNetworkPool and MultipleNetworkPool ?

Reply:
changed